2005-08-09  Gwenole Beauchesne  <gbeauchesne@mandriva.com>

	* util.c (sqlite3TextToPtr): Use an union to avoid aliasing
	issues.

--- sqlite-3.2.2/src/util.c.aliasing-fixes	2005-06-06 11:06:39.000000000 -0400
+++ sqlite-3.2.2/src/util.c	2005-08-09 07:11:17.000000000 -0400
@@ -943,7 +943,6 @@ void *sqlite3HexToBlob(const char *z){
 void *sqlite3TextToPtr(const char *z){
   void *p;
   u64 v;
-  u32 v2;
   if( z[0]=='0' && z[1]=='x' ){
     z += 2;
   }
@@ -953,11 +952,20 @@ void *sqlite3TextToPtr(const char *z){
     z++;
   }
   if( sizeof(p)==sizeof(v) ){
-    p = *(void**)&v;
+    union {
+      u64 j;
+      void *p;
+    } x;
+    x.j = v;
+    p = x.p;
   }else{
-    assert( sizeof(p)==sizeof(v2) );
-    v2 = (u32)v;
-    p = *(void**)&v2;
+    assert( sizeof(p)==sizeof(u32) );
+    union {
+      u32 i;
+      void *p;
+    } x;
+    x.i = v;
+    p = x.p;
   }
   return p;
 }
